### ISSUE TITLE
Allow users to turn off progress bar

### ISSUE THREAD

**Issue by plantarum** [torognes/vsearch#261]
Would it be possible to add a flag to turn off the progress bar for regular interactive (non-pipe) use? (as you already do automatically for pipes, i.e. #39 )

In my use-case (admittedly probably rare), I run my scripts from Emacs org-mode buffers. Emacs sends the code to run in an interactive shell, but then collects the output and inserts (variously formatted) back into the org-mode buffer. This provides a very convenient way to combine scripts and output in the same file, very similar to Python or R notebooks.

In this instance, the progress bar becomes noise in my org buffer. I can turn it off with --quiet, but then I lose useful information as well, such as merge success rates. Being able to turn it off with a --no-pipe argument would be lovely.

**Comment by frederic-mahe** [comment]
Hi, I use Emacs and org-mode but I never encountered that problem. Could you please describe your workflow? for instance, how do you capture the stderr of the org-mode buffer?

**Comment by plantarum** [comment]
Here's an excerpt from the file I'm working on now:

+-----------------------------------

#+PROPERTY: header-args:bash :session *bash* :dir /ssh:emmet:data/illumina/170616-
#re# -*- org-export-babel-evaluate: nil -*-


#+BEGIN_SRC bash :results verbatim
  # emmet wants to open dash, which won't work for us!!
  bash --norc

  # need to clear the prompt settings:
  PS1='$ '
  PS2='> '

  OUT=vs170804

  echo
  echo Ready to Begin!
#+END_SRC

#+RESULTS:
:
: $ $ $ $ $ $ $ $ > > > $ $
: Ready to Begin!

#+BEGIN_SRC bash :results output verbatim
  ## get file names, store them in samples.txt:
  RF='_R1_001.fastq';
  RR='_R2_001.fastq';

  find ./raw/ -name *$RF | xargs -n 1 basename -s $RF > $OUT/samples.txt
  echo
  echo Starting mergepairs at:
  date
  for file in `cat $OUT/samples.txt`
  do
      echo $file
      if [ -s raw/$file$RF ] ## data exists
      then
          vsearch --fastq_mergepairs raw/$file$RF \
                  --reverse raw/$file$RR \
                  --fastqout $OUT/joined/$file.joined.fq \
                  --fastqout_notmerged_fwd $OUT/joined/$file.un1.fq
                  ##--quiet 2> /dev/null  # discard messages
          cat $OUT/joined/$file.joined.fq $OUT/joined/$file.un1.fq \
            > $OUT/joined/$file.j1.fq
      else ## no data
          echo "$file" >> $OUT/joined/join.fails
      fi
  done
  echo
  echo Ending mergepairs at:
  date

 #+END_SRC

#+RESULTS:
#+begin_example

$ $ $ $ $ $ > > > > $ $
Starting mergepairs at:
Fri Aug  4 17:43:04 EDT 2017
> > > > > > > > > > > > > > > 1004_S259_L001
vsearch v2.4.3_linux_x86_64, 31.3GB RAM, 12 cores
https://github.com/torognes/vsearch
Merging reads 1%  ^MMerging reads 100%  ^MMerging reads 100%
      2301  Pairs
)
)
1008_S211_L001
vsearch v2.4.3_linux_x86_64, 31.3GB RAM, 12 cores
https://github.com/torognes/vsearch
Merging reads 1%  ^MMerging reads 100%  ^MMerging reads 100%
      2565  Pairs
)
)

+-----------------------------------

I manually added the '^M' characters in the copy above - they don't
appear in my web browser, but are present in my org buffer (colored
blue, indicating they're a non-printing character of some sort). In the
associated inferior-shell buffer, I see nice summaries like this:
Merging reads 100%
       232  Pairs
       149  Merged (64.2%)
        83  Not merged (35.8%)
913_S253_L001
vsearch v2.4.3_linux_x86_64, 31.3GB RAM, 12 cores

That's what I'd like in my org output.

Best,

Tyler

--
plantarum.ca



On Fri, Aug 4, 2017, at 06:09 PM, Frédéric Mahé wrote:
> Hi, I use Emacs and org-mode but I never encountered that problem.
> Could you please describe your workflow? for instance, how do you
> capture the stderr of the org-mode buffer?> — You are receiving this because you authored the thread. Reply to
> this email directly, view it on GitHub[1], or mute the thread[2].> 


Links:

  1. https://github.com/torognes/vsearch/issues/261#issuecomment-320363933

**Comment by plantarum** [comment]
I didn't realize, even though you mentioned it, that the output I'm trying to log is going to standard error, not standard output. I tried a few different things in my org-mode source block, to try to redirect standard error to standard output and have org insert it into the buffer as 'results'. 

I couldn't get that to work, but I did realize I could just dump standard error into a file, and link to that file, ie:

+--------------------

#+BEGIN_SRC bash :results output raw
  for file in `cat $OUT/samples.txt`
  do 
          vsearch --fastq_mergepairs raw/$file$RF \
                  --reverse raw/$file$RR \
                  --fastq_maxdiffs 10\
                  --fastqout $OUT/joined/$file.joined.fq \
                  --fastqout_notmerged_fwd $OUT/joined/$file.un1.fq 2>> $OUT/joined/join.summary
  done
#+END_SRC

[[file:vs170804/joined/join.summary][Join Summary]]

+--------------------

This is actually nicer than what I wanted in the first place, since I don't have reams of log output in my org buffer. Still not sure what I was doing wrong originally.

Best,

Tyler

**Comment by frederic-mahe** [comment]
Alternatively, you can use vsearch's `--quiet` option to suppress messages other than the merge percentages:
```sh
while read file ; do
    vsearch \
        --fastq_mergepairs raw/$file$RF \
        --reverse raw/$file$RR \
        --fastq_maxdiffs 10 \
        --fastqout $OUT/joined/$file.joined.fq \
        --fastqout_notmerged_fwd $OUT/joined/$file.un1.fq \
        --quiet
done < $OUT/samples.txt 2> $OUT/joined/join.summary
```

**Comment by torognes** [comment]
I have added a `--no_progress` option. It will not update the progress indicator until the operation is finished.

**Comment by frederic-mahe** [comment]
@torognes, note that this issue is somehow related to issue #156.

**Comment by torognes** [comment]
Fixed in version 2.5.0.
